cmd/tsbs_generate_queries/databases/redistimeseries/devops.go

```diff
@@ -0,0 +1,145 @@
+package redistimeseries
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/timescale/tsbs/cmd/tsbs_generate_queries/uses/devops"
+	"github.com/timescale/tsbs/query"
+)
+
+// TODO: Remove the need for this by continuing to bubble up errors
+func panicIfErr(err error) {
+	if err != nil {
+		panic(err.Error())
+	}
+}
+
+const (
+	oneMinuteMillis = 60 * 1000
+	fiveMinuteMillis = 5 * oneMinuteMillis
+	oneHourMillis   = oneMinuteMillis * 60
+)
+
+// Devops produces RedisTimeSeries-specific queries for all the devops query types.
+type Devops struct {
+	*BaseGenerator
+	*devops.Core
+}
+
+// GenerateEmptyQuery returns an empty query.RedisTimeSeries
+func (d *Devops) GenerateEmptyQuery() query.Query {
+	return query.NewRedisTimeSeries()
+}
+
+// GroupByTime fetches the MAX for numMetrics metrics under 'cpu', per minute for nhosts hosts,
+// every 5 mins for 1 hour
+func (d *Devops) GroupByTime(qi query.Query, nHosts, numMetrics int, timeRange time.Duration) {
+	interval := d.Interval.MustRandWindow(timeRange)
+
+	redisQuery := fmt.Sprintf(`TS.MRANGE %d %d AGGREGATION max %d FILTER measurement=cpu`,
+		interval.StartUnixMillis(),
+		interval.EndUnixMillis(),
+		fiveMinuteMillis)
+
+	metrics, err := devops.GetCPUMetricsSlice(numMetrics)
+	panicIfErr(err)
+	// we only need to filter if we we dont want all of them
+	if numMetrics != devops.GetCPUMetricsLen() {
+	redisQuery += " fieldname="
+	if numMetrics > 1 {
```

Comment:
I really dont see the necessity out of it.
Performance wise it will lead to the same place. and this does not influences our benchmarks since its done on a pre-step. 
benchmark values at:
https://gist.github.com/dtjm/c6ebc86abe7515c988ec

I agree with this comment:
> strings.join is not as a general replacement for + or strings.concat, but in being the "reverse operation" of a strings.split operation. It makes more sense in that context - when you have a bunch of strings that you want to join together using a delimiter - than as a replacement for concat.